lib/thrift/generator/client.ex

```diff
@@ -0,0 +1,275 @@
+defmodule Thrift.Generator.Client do
+  defmodule Framed do
+    alias Thrift.Generator.Service
+    alias Thrift.Parser.Models.Function
+
+    def generate(service_module, service) do
+      functions = service.functions
+      |> Map.values
+      |> Enum.map(&generate_function(service_module, &1))
+
+      quote do
+        defmodule Client.Framed do
+          alias Thrift.Protocols.Binary
+          alias Thrift.Exceptions.TApplicationException
+
+          defmodule State do
+            defstruct host: nil, port: nil, tcp_opts: nil, timeout: 5000, sock: nil
+          end
+
+          use Connection
+
+          def init({host, port, tcp_opts, timeout}) do
+            s = %State{host: host,
+                       port: port,
+                       tcp_opts: tcp_opts,
+                       timeout: timeout,
+                       sock: nil}
+
+            {:connect, :init, s}
+          end
+
+          def start_link(host, port, tcp_opts, timeout \\ 5000) do
+            Connection.start_link(__MODULE__, {host, port, tcp_opts, timeout})
+          end
+
+          def send(conn, data), do: Connection.call(conn, {:send, data})
+
+          def recv(conn, bytes, timeout \\ 5000) do
+            Connection.call(conn, {:recv, bytes, timeout})
+          end
+
+          def close(conn), do: Connection.call(conn, :close)
+
+          def connect(_, %{sock: nil, host: host, port: port, tcp_opts: opts, timeout: timeout} = s) do
+            case :gen_tcp.connect(host, port, [active: false, packet: 4] ++ opts, timeout) do
```

Comment:
Probably want to `Keyword.put_new` a `:send_timeout`, default is `:infinity`. That is time to wait to pass the data to the kernel's sndbuffer.